src/app/views/home/home-view.tsx

```diff
@@ -379,12 +388,19 @@ export default function HomeView() {
   };
 
   const onToggleFullscreen = async () => {
-    if (typeof document === "undefined") return;
-
-    if (!document.fullscreenElement) {
-      await fullscreenRef.current?.requestFullscreen?.();
-    } else {
-      await document.exitFullscreen?.();
+    const el = fullscreenRef.current;
+    if (!el) return;
+
+    try {
+      if (!isFullscreen) {
+        await requestFullscreen(el);
+      } else {
+        await exitFullscreen();
+      }
+
+      setIsFullscreen(checkFullscreen());
```

Comment:
Again, IMO, you dont need to set that state here. `requestFullscreen` and `exitFullscreen` would trigger the 
```
    document.addEventListener("fullscreenchange", onFullscreenChange);
    document.addEventListener("webkitfullscreenchange", onFullscreenChange); // Safari / Mac
    window.addEventListener("resize", onResize);
```
Which will set the state in the event handler